doc/compiler/optimizer/IntroLoopOptimizations.md

```diff
@@ -0,0 +1,183 @@
+
+```
+                    Fast/Hot Loop (A)                                       Slow/Cold Loop (A')
+                               |                                                       |
+                               |                                                       |
+                               v                                                       v
+                            Guard (G)                                               Guard (G')
+                            /      |                                                /     \
+                           /       |                                               /       \
+                          v        |                                              v         v
+ Inlined code (fast path) (I)      |                     Inlined code (fast path) (I')     call (slow path) (C')
+                                   |                                                                ^
+                                   |                                                                |
+                                   +----------------------------------------------------------------+
+
+```
+The point of this optimization is to essentially remove the call from the fast loop by sending the slow path out of that guard in the fast/hot loop into the slow loop. After the transformation, the fast loop only has the fast path. It has the guard and its fast path, but if the guard fails, it goes into the slow loop and take the slow path (`C’`) in the slow loop. The slow path (`C`) is eliminated from the fast loop. When the call (`C`) is gone, it may open up opportunities for other optimizations in the fast loop.
+
+The bad thing about Loop Transfer is that it barges into the slow loop. It does not enter the slow loop in a well-behaved way. A loop should be entered from the entry block of the loop. That is the main property of a natural loop structure. A loop should not be entered at any place. It should be entered in a nice and structured way at the top of the loop.
+
+By changing the guard `G` to barge into `C’` in the middle of the slow loop without going through the entry block, it is no longer a well-behaved loop. By doing so, an improper region structure is created. An improper region indicates something is going on with the loop and it is no longer a well-structured control flow. Other optimizations become conservative around improper regions. If an improper region is created, the loop is not versioned, unrolled, loop stride on, and even VP forgets all the constraints it knows about at that time.
+
+### Avoid Creation of Improper Regions in Loop Transfer ([Work in Progress](https://github.com/eclipse-openj9/openj9/issues/13243))
```

Comment:
Instead of "Work in Progress" we can perhaps just say "#13243". This way we don't have to remember to update this document when the work is delivered.

Reply:
Addressed in 641a90a